packages/material-ui-lab/src/PickersDay/PickersDay.tsx

```diff
@@ -2,82 +2,130 @@ import * as React from 'react';
 import {
-  StyleRules,
-  MuiStyles,
   useTheme,
-  WithStyles,
-  withStyles,
   alpha,
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
 } from '@material-ui/core/styles';
 import { useForkRef } from '@material-ui/core/utils';
 import { ExtendMui } from '../internal/pickers/typings/helpers';
 import { useUtils } from '../internal/pickers/hooks/useUtils';
 import { DAY_SIZE, DAY_MARGIN } from '../internal/pickers/constants/dimensions';
 import { PickerSelectionState } from '../internal/pickers/hooks/usePickerState';
+import pickersDayClasses, { getPickersDayUtilityClass } from './pickersDayClasses';
 
-export type PickersDayClassKey =
-  | 'root'
-  | 'dayWithMargin'
-  | 'dayOutsideMonth'
-  | 'hiddenDaySpacingFiller'
-  | 'today'
-  | 'selected'
-  | 'disabled';
+export type PickersDayClassKey = keyof typeof pickersDayClasses;
 
-export const styles: MuiStyles<PickersDayClassKey> = (theme): StyleRules<PickersDayClassKey> => ({
-  root: {
-    ...theme.typography.caption,
-    width: DAY_SIZE,
-    height: DAY_SIZE,
-    borderRadius: '50%',
-    padding: 0,
-    // background required here to prevent collides with the other days when animating with transition group
-    backgroundColor: theme.palette.background.paper,
-    color: theme.palette.text.primary,
-    '&:hover': {
-      backgroundColor: alpha(theme.palette.action.active, theme.palette.action.hoverOpacity),
-    },
-    '&:focus': {
-      backgroundColor: alpha(theme.palette.action.active, theme.palette.action.hoverOpacity),
-      '&$selected': {
-        willChange: 'background-color',
-        backgroundColor: theme.palette.primary.dark,
-      },
-    },
-    '&$selected': {
-      color: theme.palette.primary.contrastText,
-      backgroundColor: theme.palette.primary.main,
-      fontWeight: theme.typography.fontWeightMedium,
-      transition: theme.transitions.create('background-color', {
-        duration: theme.transitions.duration.short,
-      }),
-      '&:hover': {
-        willChange: 'background-color',
-        backgroundColor: theme.palette.primary.dark,
-      },
+const useUtilityClasses = (styleProps: PickersDayProps<any>) => {
+  const {
+    selected,
+    disableMargin,
+    disableHighlightToday,
+    today,
+    outsideCurrentMonth,
+    showDaysOutsideCurrentMonth,
+    classes,
+  } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      selected && 'selected',
+      !disableMargin && 'dayWithMargin',
+      !disableHighlightToday && today && 'today',
+      outsideCurrentMonth && showDaysOutsideCurrentMonth && 'dayOutsideMonth',
+    ],
+    hiddenDaySpacingFiller: ['hiddenDaySpacingFiller'],
+  };
+
+  return composeClasses(slots, getPickersDayUtilityClass, classes);
+};
+
+const PickersDayRoot = experimentalStyled(
+  ButtonBase,
+  {},
+  {
+    name: 'MuiPickersDay',
+    slot: 'Root',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
```

Comment:
```suggestion
      const { styleProps = {} } = props;
```

And we can avoid using `styleProps?` everywhere. Can it really be undefined tough? Is typescript complaining?